src/thor/pathalgorithm.cc

```diff
@@ -0,0 +1,128 @@
+#include "thor/pathalgorithm.h"
+#include "baldr/datetime.h"
+#include "midgard/logging.h"
+#include "sif/edgelabel.h"
+
+#include <chrono>
+#include <ctime>
+
+namespace dt = valhalla::baldr::DateTime;
+namespace sc = std::chrono;
+
+namespace valhalla {
+namespace thor {
+
+// helper function to initialize the object from a location
+TimeInfo TimeInfo::make(valhalla::Location& location, baldr::GraphReader& reader) {
```

Comment:
so this is where we instantiate a TimeInfo it checks if the request has time information (via the location), gets the timezone (so we can do offsetting and localtime), then it gets the timezone offset local time in epoch seconds, if it was a current time route it sets the time to the current time at that timezone, it also figures out what second of the week we are at and how many seconds we are away from the current time. so if you do depart at 5 minutes ago,  it can track your time as you cross over "now" which is useful for current/realtime traffic flow enabled requests